File: web/src/api/normal/gost_client_proxy.js
import request from "../../request/index.js";

const baseUrl = '/v1/normal/gost/client/proxy'

export const apiNormalGostClientProxyList = () => {
    return request.request({
        url: `${baseUrl}/list`,
        method: 'POST',
    })
}

export const apiNormalGostClientProxyPage = (data) => {
    return request.request({
        url: `${baseUrl}/page`,
        method: 'POST',
        data
    })
}

export const apiNormalGostClientProxyCreate = (data) => {
    return request.request({
        url: `${baseUrl}/create`,
        method: 'POST',
        data
    })
}

export const apiNormalGostClientProxyRenew = (data) => {
    return request.request({
        url: `${baseUrl}/renew`,
        method: 'POST',
        data
    })
}

export const apiNormalGostClientProxyEnable = (data) => {
    return request.request({
        url: `${baseUrl}/enable`,
        method: 'POST',
        data
    })
}

export const apiNormalGostClientProxyUpdate = (data) => {
    return request.request({
        url: `${baseUrl}/update`,
        method: 'POST',
        data
    })
}

export const apiNormalGostClientProxyDelete = (data) => {
    return request.request({
        url: `${baseUrl}/delete`,
        method: 'POST',
        data
    })
}

File: server/router/normal/gost_client_proxy.go
package normal

import (
	"github.com/gin-gonic/gin"
	"server/controller/normal/gost_client_proxy"
	"server/global"
	"server/router/middleware"
)

func InitGostClientProxy(group *gin.RouterGroup) {
	g := group.Group("gost/client/proxy", middleware.Auth(global.Jwt))
	g.POST("create", gost_client_proxy.Create)
	g.POST("renew", gost_client_proxy.Renew)
	g.POST("update", gost_client_proxy.Update)
	g.POST("enable", gost_client_proxy.Enable)
	g.POST("delete", gost_client_proxy.Delete)
	g.POST("page", gost_client_proxy.Page)
}
